File: python/problems/binary_search.py
from environment import TestCase, Problem

BINARY_SEARCH_PROBLEM = Problem(
    name="Binary Search",
    description=(
        "Given a sorted array of integers and a target value, return the index of the"
        "target if it exists in the array, otherwise return -1."
    ),
    test_cases=[
        TestCase(input_data={"lista": [-1,0,3,5,9,12], "target": 9}, expected_output=4),
        TestCase(input_data={"lista": [-1,0,3,5,9,12], "target": 2}, expected_output=-1),
        TestCase(input_data={"lista": [-1,0,3,5,9,12], "target": 3}, expected_output=2),
        # Larger test cases
        TestCase(input_data={"lista": [1,2,3,4,5,6,7,8,9,10], "target": 10}, expected_output=9),
    ],
)

File: go/problems/binary_search.go
package problems

import env "algorithms/environment"

var BinarySearch = env.Problem{
	Name: "Binary Search",
	TestCases: []env.TestCase{
		env.TestCase{InputData: map[string]any{"arr": []int{1, 2, 3, 4, 5}, "target": 3}, ExpectedOutput: 2},
		env.TestCase{InputData: map[string]any{"arr": []int{1, 2, 3, 4, 5}, "target": 6}, ExpectedOutput: -1},
		env.TestCase{InputData: map[string]any{"arr": []int{1, 2, 3, 4, 5}, "target": 1}, ExpectedOutput: 0},
		env.TestCase{InputData: map[string]any{"arr": []int{1, 2, 3, 4, 5}, "target": 5}, ExpectedOutput: 4},
	},
}
